extractor/extractor.py

```diff
@@ -0,0 +1,195 @@
+import re
+from urllib.request import urlopen
+from pprint import pprint
+
+import requests
+import bs4
+from bs4 import BeautifulSoup, Comment
+from PIL import Image
+
+import contents
+from contents import ContentType
+
+req_session = ""
+all_text_tags = {'p', 'span', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'code'}
+heading_tags = {'h1', 'h2', 'h3', 'h4', 'h5', 'h6'}
+pattern_for_video_ext = re.compile(r'^.*\.(mp4|ogg|mov|webm)')
+pattern_for_ads = re.compile('(^ad-|^ads-)')
+
+class Extractor:
+    def __init__(self, URL):
+        self.url = URL
+        self.soup = None
+        self.contents_list = contents.Contents()
+
+    def extract_html(self):
+        global req_session
+
+        req_session = requests.Session()
+        file = req_session.get(self.url).text    # To request Html from URL
+
+        # Remove the comment to read local files for testing purpose
+        # file = open('./test_html.html','r').read()
+
+        self.soup = BeautifulSoup(file,'lxml')
+        self.clean_soup()
+
+        self.extract_data_from_html()
+
+    def clean_soup(self):
+        decomposable_tags = self.soup.find_all(['script', 'noscript', 'style', 'header', 'footer', 'meta', 'link'])
+        comments = self.soup.find_all(text=lambda text: isinstance(text, Comment))
+        ads = self.soup.find_all(re.compile('.*'), {'class': pattern_for_ads})
+
+        comments = [comment.extract() for comment in comments]
+        [tag.decompose() for tag in decomposable_tags]
+        [ad.decompose() for ad in ads]
+
+    def extract_data_from_html(self):
+        # Extract data from head
+        head = self.soup.find('head')
+        self.extract_data_from_html_head(head)
+
+        # Extract data from body
+        body_soup = self.soup.find('body')
+        self.extract_data_from_html_body(body_soup)
+
+    def extract_data_from_html_head(self, soup_head):
+        title = soup_head.title
+        text_type = 'title'
+        text_string = title.get_text()
+        _content = contents.Text(text_string, text_type)
+        self.contents_list.add_content(_content)
+
+    def extract_data_from_html_body(self, node):
+        isContent, _content = self.is_content(node)
+
+        # Add to the final list if a valid content is extracted
+        if isContent:
+            self.contents_list.add_content(_content)
+
+        if node is not None and node.name is not None and not isContent:
```

Comment:
`if node and node.name and not isContent`

Reply:
Updated! Thanks for the suggestion.